okhttp-client-handler/monix/src/main/scala/com/softwaremill/sttp/okhttp/monix/OkHttpMonixClientHandler.scala

```diff
@@ -0,0 +1,72 @@
+package com.softwaremill.sttp.okhttp.monix
+
+import java.nio.ByteBuffer
+
+import com.softwaremill.sttp._
+import com.softwaremill.sttp.okhttp.OkHttpAsyncClientHandler
+import monix.eval.Task
+import monix.execution.{Cancelable, Scheduler}
+import monix.reactive.{Consumer, Observable}
+import okhttp3.{MediaType, OkHttpClient, RequestBody => OkHttpRequestBody}
+import okio.BufferedSink
+
+import scala.language.higherKinds
+import scala.util.{Failure, Success, Try}
+
+/**
+  * Created by omainegra on 8/4/17.
+  */
+class OkHttpMonixClientHandler private (client: OkHttpClient)(
+    implicit s: Scheduler)
+    extends OkHttpAsyncClientHandler[Task, Observable[ByteBuffer]](client,
+                                                                   TaskMonad) {
+
+  private lazy val io = Scheduler.io("sttp-monix-io")
+
+  override def streamToRequestBody(
+      stream: Observable[ByteBuffer]): Option[OkHttpRequestBody] =
+    Some(new OkHttpRequestBody() {
+      override def writeTo(sink: BufferedSink): Unit =
+        stream
+          .consumeWith(
+            Consumer.foreach(chunk => sink.write(chunk.array()))
+          )
+          .runAsync(io)
```

Comment:
I'm not sure this is going to work correctly. Looking at [the example](https://github.com/square/okhttp/blob/master/samples/guide/src/main/java/okhttp3/recipes/PostStreaming.java) from OkHttp, the `writeTo` method is **synchronous** - that is, it should only complete when the entire stream has been written. Otherwise, OkHttp has no way to tell if the request is finished or not.

This is still useful - however I think it should be changed to run synchronously and clearly marked in the docs that it might block the calling thread (the thread that's calling `send()`).